### PR TITLE
Add gremlin-dump test

### PR DESCRIPTION
**Pull request by eonpatapon**

### CODE REVIEW

**Comment by eonpatapon** [comment]
http://84.39.63.212/eval/58

File: test/gremlin-dump.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ pkgs
+, contrailPkgs
+, stdenv
+}:
+
+with import (pkgs.path + /nixos/lib/testing.nix) { system = builtins.currentSystem; };
+
+let
+
+  dump = stdenv.mkDerivation {
```
Comment (nlewo) [inline test/gremlin-dump.nix]: Do you really need a derivation for this? Is it not sufficient to just `cassandraDumpPath = ./cassandra-dump.tgz` ?

File: test/gremlin-dump.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ];
+    config = {
+      _module.args = { inherit contrailPkgs; };
+
+      contrail.databaseLoader = {
+        enable = true;
+        cassandraDumpPath = dump;
+      };
+
+      gremlin.server = {
```
Comment (nlewo) [inline test/gremlin-dump.nix]: `gremlin.server.enable = true;`

File: modules/gremlin-server.nix [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        after = [ "network.target" ] ++ (pkgs.lib.optional cfg.makeDump "gremlinDump.service");
+        requires = pkgs.lib.optional cfg.makeDump "gremlinDump.service";
+        script = "${contrailPkgs.tools.gremlinServer}/bin/gremlin-server ${serverConf}";
+      };
+    })
+    (mkIf cfg.makeDump {
+      systemd.services.gremlinDump = {
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" "cassandra.service" ];
+        script = "${contrailPkgs.tools.contrailGremlin}/bin/gremlin-dump --cassandra 127.0.0.1:9042 ${cfg.dumpPath}";
```
[the remark this answers is not in the document]
Comment (eonpatapon) [reply modules/gremlin-server.nix]: It works well, and I will have another test in nixpkgs-cloudwatt that use this feature

File: test/gremlin-dump.nix (hunk 1 above)
Comment (eonpatapon) [reply test/gremlin-dump.nix]: No, it was completely broken in your test

The module expect a path where it can find dump files, not a .tgz

**Pull request merged**